src/cstringholder.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::ffi::CStr;
+use std::ops::Deref;
+use std::os::raw::c_char;
+use std::ptr;
+use translate::*;
+use types::{StaticType, Type};
+
+use gobject_ffi;
+use ffi as glib_ffi;
+use value::{FromValueOptional, Value};
+
+#[derive(Debug, Hash, PartialOrd, Ord)]
+pub struct CStringHolder {
+    slice: Box<CStr>,
+    ptr: *const c_char,
+    owned: bool,
```

Comment:
I would only do this for owned strings, otherwise it doesn't not really make sense or does it? Do you have an example usage for a non-owned string?

Reply:
In the unit-tests of the gstreamer subcrate I had to manually create CStringHolders.